tendermint-testgen/src/commit.rs

```diff
@@ -0,0 +1,159 @@
+use gumdrop::Options;
+use serde::Deserialize;
+use simple_error::*;
+use tendermint::{ block, lite };
+
+use crate::{Generator, Validator, Header, Vote, helpers::*};
+
+#[derive(Debug, Options, Deserialize, Clone)]
+pub struct Commit {
+    #[options(help = "header (required)", parse(try_from_str = "parse_as::<Header>"))]
+    pub header: Option<Header>,
+    #[options(help = "votes in this commit (default: from header)",
+    parse(try_from_str = "parse_as::<Vec<Vote>>"))]
+    pub votes: Option<Vec<Vote>>,
+    #[options(help = "commit round (default: 1)")]
+    pub round: Option<u64>
+}
+
+impl Commit {
+    pub fn new(header: &Header) -> Self {
+        Commit {
+            header: Some(header.clone()),
```

Comment:
Usually preferable to leave it up to the callee to clone the data passed to a function. This can avoid unnecessary clones in cases where the callee already has ownership of the data passed to the function.

```suggestion
    pub fn new(header: Header) -> Self {
        Commit {
            header: Some(header),
```

Reply:
thanks:) yes; ownership and borrowing is one of the rusty things I still struggle with...